src/Cache/Persister/Entity/AbstractEntityPersister.php

```diff
@@ -611,12 +616,13 @@ public function refresh(array $id, $entity, $lockMode = null)
      *
      * @return CollectionCacheKey
      */
-    protected function buildCollectionCacheKey(array $association, $ownerId)
+    protected function buildCollectionCacheKey(array $association, $ownerId, string $filterHash)
```

Comment:
@greg0ire Is introducing a new argument to a protected method okay in a version-branch that's already released? It may break projects that are implementing the Abstract class.

Reply:
I didn't spot that one. You're right, it's a breaking change for extending classes, that suddenly need to have that extra argument, and for callers, that need to provide it.

@dbannik please resort to `func_get_args()`. The new argument should be added in the next major release.

Reply:
@greg0ire  instead of `func_get_args` can we use default value ?
```suggestion
    protected function buildCollectionCacheKey(array $association, $ownerId, string $filterHash = '')
````

Reply:
That would still be a breaking change for extending classes (they would suddenly have incompatible signatures).

Reply:
resolved